src/azure-cli/azure/cli/command_modules/acr/task.py

```diff
@@ -224,10 +229,17 @@ def acr_task_create(cmd,  # pylint: disable=too-many-locals
     )
 
     try:
-        return client.create(resource_group_name=resource_group_name,
-                             registry_name=registry_name,
-                             task_name=task_name,
-                             task_create_parameters=task_create_parameters)
+        # BUG: Discriminator type is absent or null, use base class TaskStepProperties.
+        #      Deserialized twice in SDK: `deserialized = self._deserialize('Task', pipeline_response)`
+        def local_fix(pipeline_response, deserialized, _):
+            if pipeline_response.http_response.status_code == 200:
+                return deserialized
+            return pipeline_response
+        return client._create_initial(resource_group_name=resource_group_name,  # pylint: disable=protected-access
```

Comment:
>_create_initial [](http://example.com/codeflow?start=22&length=15)

I'm curious why you use private method `_create_initial` instead of public method `begin_create`?

Reply:
As the bug I said before, `begin_create` deserializes pipeline_response twice which will change the `pipeline_response.content`. It makes the deserialized data wrong returned by `begin_create`.
So I have to get the data deserialized at first time in `_create_initial`.